geocoder/index.cpp

```diff
@@ -88,29 +91,47 @@ void Index::AddStreet(DocId const & docId, Index::Doc const & doc)
 
 void Index::AddHouses()
 {
-  size_t numIndexed = 0;
-  for (DocId docId = 0; docId < static_cast<DocId>(m_docs.size()); ++docId)
-  {
-    auto const & buildingDoc = GetDoc(docId);
+  atomic<size_t> numIndexed{0};
+  std::mutex mutex;
 
-    if (buildingDoc.m_type != Type::Building)
-      continue;
+  vector<thread> threads(thread::hardware_concurrency());
```

Comment:
Please add a CHECK that threads is not empty after this.

Reply:
done